Index: src/main/java/com/socialmeli/socialmeli/services/UserService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.socialmeli.socialmeli.services;\n\nimport org.springframework.stereotype.Service;\n\n@Service\npublic class UserService implements IUserService{\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/socialmeli/socialmeli/services/UserService.java b/src/main/java/com/socialmeli/socialmeli/services/UserService.java
--- a/src/main/java/com/socialmeli/socialmeli/services/UserService.java	
+++ b/src/main/java/com/socialmeli/socialmeli/services/UserService.java	
@@ -1,7 +1,27 @@
 package com.socialmeli.socialmeli.services;
 
+import com.socialmeli.socialmeli.dto.UserDto;
+import com.socialmeli.socialmeli.dto.UserFollowedDto;
+import com.socialmeli.socialmeli.repositories.IUserRepository;
 import org.springframework.stereotype.Service;
+import java.util.List;
 
 @Service
 public class UserService implements IUserService{
+
+    IUserRepository userRepository;
+
+    @Override
+    public UserFollowedDto listFollowed(Integer userId){
+
+        List<UserDto> listFollowed = userRepository.listFollowed(userId);
+
+        UserDto userDto = userRepository.getUserDetails(userId);
+
+        if(userDto != null){
+            return new UserFollowedDto(userDto.user_id(), userDto.user_name(), listFollowed);
+        }else{
+            return null;
+        }
+    }
 }
Index: src/main/java/com/socialmeli/socialmeli/repositories/IUserRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.socialmeli.socialmeli.repositories;\n\nimport com.socialmeli.socialmeli.entities.User;\n\nimport java.util.ArrayList;\n\npublic interface IUserRepository extends ICrudRepository{\n    ArrayList<User> loadUserJson();\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/socialmeli/socialmeli/repositories/IUserRepository.java b/src/main/java/com/socialmeli/socialmeli/repositories/IUserRepository.java
--- a/src/main/java/com/socialmeli/socialmeli/repositories/IUserRepository.java	
+++ b/src/main/java/com/socialmeli/socialmeli/repositories/IUserRepository.java	
@@ -1,9 +1,17 @@
 package com.socialmeli.socialmeli.repositories;
 
+import com.socialmeli.socialmeli.dto.UserDto;
 import com.socialmeli.socialmeli.entities.User;
+import org.springframework.stereotype.Repository;
 
+import java.util.List;
 import java.util.ArrayList;
 
+@Repository
 public interface IUserRepository extends ICrudRepository{
     ArrayList<User> loadUserJson();
+
+    List<UserDto> listFollowed(Integer userId);
+
+    UserDto getUserDetails(Integer userId);
 }
Index: src/main/java/com/socialmeli/socialmeli/repositories/UserRepositoryImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.socialmeli.socialmeli.repositories;\n\nimport com.socialmeli.socialmeli.entities.Post;\nimport com.socialmeli.socialmeli.entities.User;\n\nimport java.util.ArrayList;\nimport java.util.Optional;\n\npublic class UserRepositoryImpl implements IUserRepository{\n    ArrayList<User> users;\n\n    public UserRepositoryImpl() {\n        this.users = this.loadUserJson();\n    }\n\n    @Override\n    public Object save(Object o) {\n        return null;\n    }\n\n    @Override\n    public Object update(Object o) {\n        return null;\n    }\n\n    @Override\n    public Boolean deleteById(Integer id) {\n        return null;\n    }\n\n    @Override\n    public Optional<User> findById(Integer id) {\n        return Optional.empty();\n    }\n\n    @Override\n    public ArrayList<User> findAll() {\n        return null;\n    }\n\n    @Override\n    public ArrayList<User> loadUserJson() {\n        return null;\n    }\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/socialmeli/socialmeli/repositories/UserRepositoryImpl.java b/src/main/java/com/socialmeli/socialmeli/repositories/UserRepositoryImpl.java
--- a/src/main/java/com/socialmeli/socialmeli/repositories/UserRepositoryImpl.java	
+++ b/src/main/java/com/socialmeli/socialmeli/repositories/UserRepositoryImpl.java	
@@ -1,10 +1,11 @@
 package com.socialmeli.socialmeli.repositories;
 
-import com.socialmeli.socialmeli.entities.Post;
+import com.socialmeli.socialmeli.dto.UserDto;
 import com.socialmeli.socialmeli.entities.User;
 
-import java.util.ArrayList;
-import java.util.Optional;
+
+import java.util.*;
+import java.util.stream.Collectors;
 
 public class UserRepositoryImpl implements IUserRepository{
     ArrayList<User> users;
@@ -42,4 +43,42 @@
     public ArrayList<User> loadUserJson() {
         return null;
     }
+
+    @Override
+    public List<UserDto> listFollowed(Integer userId){
+
+        User user = users.stream()
+                .filter(u -> u.getUserId().equals(userId))
+                .findFirst()
+                .orElse(null);
+
+        if(user != null && user.getFollowed() != null){
+
+            return user.getFollowed().stream()
+                    .map(this::converToUserDto)
+                    .collect(Collectors.toList());
+        }else{
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public UserDto getUserDetails(Integer userId){
+
+        User user = users.stream()
+                .filter(u -> u.getUserId().equals(userId))
+                .findFirst()
+                .orElse(null);
+
+        if(user != null){
+            return new UserDto(user.getUserId(), user.getUserName());
+        }else{
+            return null;
+        }
+    }
+
+    private UserDto converToUserDto(User user){
+        return new UserDto(user.getUserId(), user.getUserName());
+    }
+
 }
Index: src/main/java/com/socialmeli/socialmeli/controller/SocialController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.socialmeli.socialmeli.controller;\n\nimport org.springframework.web.bind.annotation.RestController;\n\n@RestController\npublic class SocialController {\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/socialmeli/socialmeli/controller/SocialController.java b/src/main/java/com/socialmeli/socialmeli/controller/SocialController.java
--- a/src/main/java/com/socialmeli/socialmeli/controller/SocialController.java	
+++ b/src/main/java/com/socialmeli/socialmeli/controller/SocialController.java	
@@ -1,7 +1,26 @@
 package com.socialmeli.socialmeli.controller;
 
+import com.socialmeli.socialmeli.dto.UserFollowedDto;
+import com.socialmeli.socialmeli.entities.User;
+import com.socialmeli.socialmeli.services.IUserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.ArrayList;
+import java.util.List;
+
+
 @RestController
 public class SocialController {
+
+     IUserService userService;
+
+    @GetMapping("/users/{userId}/followed/list")
+    public ResponseEntity<UserFollowedDto> getAllFollowed(@PathVariable Integer userId){
+        return new ResponseEntity<>(userService.listFollowed(userId), HttpStatus.OK);
+    }
+
 }
Index: src/main/java/com/socialmeli/socialmeli/services/IUserService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.socialmeli.socialmeli.services;\n\npublic interface IUserService {\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/socialmeli/socialmeli/services/IUserService.java b/src/main/java/com/socialmeli/socialmeli/services/IUserService.java
--- a/src/main/java/com/socialmeli/socialmeli/services/IUserService.java	
+++ b/src/main/java/com/socialmeli/socialmeli/services/IUserService.java	
@@ -1,4 +1,9 @@
 package com.socialmeli.socialmeli.services;
 
+import com.socialmeli.socialmeli.dto.UserFollowedDto;
+
 public interface IUserService {
+
+    UserFollowedDto listFollowed(Integer userId);
+
 }
Index: src/main/java/com/socialmeli/socialmeli/entities/User.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.socialmeli.socialmeli.entities;\n\nimport lombok.AllArgsConstructor;\nimport lombok.Getter;\nimport lombok.NoArgsConstructor;\nimport lombok.Setter;\n\nimport java.util.List;\n\n@Getter\n@Setter\n@AllArgsConstructor\n@NoArgsConstructor\npublic class User {\n    private int userId;\n    private String userName;\n    private List<User> followers;\n    private List<User> followed;\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/socialmeli/socialmeli/entities/User.java b/src/main/java/com/socialmeli/socialmeli/entities/User.java
--- a/src/main/java/com/socialmeli/socialmeli/entities/User.java	
+++ b/src/main/java/com/socialmeli/socialmeli/entities/User.java	
@@ -12,7 +12,7 @@
 @AllArgsConstructor
 @NoArgsConstructor
 public class User {
-    private int userId;
+    private Integer userId;
     private String userName;
     private List<User> followers;
     private List<User> followed;
